src/Opw.PineBlog.Core/Posts/UnpublishPostCommand.cs

```diff
@@ -41,18 +40,14 @@ public Handler(IBlogEntityDbContext context)
             /// <param name="cancellationToken">A cancellation token.</param>
             public async Task<Result<Post>> Handle(UnpublishPostCommand request, CancellationToken cancellationToken)
             {
-                var entity = await _context.Posts.SingleOrDefaultAsync(e => e.Id.Equals(request.Id));
-                if (entity == null)
-                    return Result<Post>.Fail(new NotFoundException<Post>($"Could not find post, id: \"{request.Id}\""));
-
-                entity.Published = null;
-                
-                _context.Posts.Update(entity);
-                var result = await _context.SaveChangesAsync(true, cancellationToken);
-                if (!result.IsSuccess)
-                    return Result<Post>.Fail(result.Exception);
-
-                return Result<Post>.Success(entity);
+                try
+                {
+                    return await _repo.UnpublishPostAsync(request.Id, cancellationToken);
+                }
+                catch (Exception ex)
```

Comment:
Only catch specific exceptions, the rest should just be thrown. That was why I didn't put any try catch blocks in the code.